mrp_schedule_state/mrp.py

```diff
@@ -77,10 +81,35 @@ def _get_values_from_selection(self, field):
 
     @api.multi
     def write(self, vals, update=True):
-        if vals.get('schedule_state') == 'scheduled':
-            vals['date_planned'] = fields.Datetime.now()
+        if vals.get('schedule_state') == 'scheduled' and not \
+                vals.get('schedule_date'):
+            vals['schedule_date'] = fields.Datetime.now()
+        if vals.get('schedule_state') and \
+                vals.get('schedule_state') != 'scheduled':
+            vals['schedule_date'] = False
         return super(MrpProduction, self).write(vals, update=update)
 
+    @api.multi
+    def set_planable_mo(self):
+        """ Set the MO as to able to be manufactured 'ToDo'
+            if it is ready to produce
+        """
+        for mo in self:
+            # If MO has been scheduled when it was not ready yet (still in
+            # waiting schedule_state, we can jump to schedule state already)
+            if mo.date_planned:
+                new_state = 'scheduled'
+            else:
+                new_state = 'todo'
+            mo.write({'schedule_state': new_state})
```

Comment:
why not `mo.schedule_state = new_state` ?

Reply:
Oui, ca aurait pu, c'est pareil